dotCMS/src/main/java/com/dotcms/ai/api/OpenAITranslationService.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    @Override
+    public List<ServiceParameter> getServiceParameters() {
+        throw new DotRuntimeException("Not implemented");
+    }
+
+    @Override
+    public void setServiceParameters(List<ServiceParameter> params, String hostId) {
+        throw new DotRuntimeException("Not implemented");
+    }
+
+    @Override
+    public List<Contentlet> translateContent(Contentlet contentlet, List<Language> langs,
+            List<com.dotmarketing.portlets.structure.model.Field> oldFields,
+            User user) throws TranslationException {
+
+        return langs.stream().map(lang -> {
+                    try {
+                        return translateContent(contentlet, lang, oldFields, user);
+                    } catch (TranslationException e) {
+                        Logger.warnAndDebug(OpenAITranslationService.class, e.getMessage(), e);
+                        return null;
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+
+    }
+
+    @Override
+    public Contentlet translateContent(Contentlet contentlet, Language targetLanguage, List<com.dotmarketing.portlets.structure.model.Field> oldFields, User user)
+            throws TranslationException {
+        Language sourceLang = APILocator.getLanguageAPI().getLanguage(contentlet.getLanguageId());
+        List<Field> fields = new LegacyFieldTransformer(oldFields).asList();
+        JSONObject sourceJson = new JSONObject();
+        fields.forEach(f -> {
+
+
+            String value = contentlet.getStringProperty(f.variable());
+            if (UtilMethods.isSet(value)) {
+                if(StringUtils.isJson(value)){
+                    sourceJson.put(f.variable(), new JSONObject(value));
+                }else{
+                    sourceJson.put(f.variable(), value);
+                }
+
+            }
+        });
+
+        Optional<String> translationKeyPrefix = Optional.ofNullable((String) contentlet.getMap().get(
+                OpenAITranslationActionlet.TRANSLATION_KEY_PREFIX));
+
+        JSONObject translationKeysJSON = new JSONObject(getTranslationKeys(translationKeyPrefix, contentlet.getLanguageId(), targetLanguage.getId()));
+
+        String systemPromptTemplate = getAISystemTranslationPrompt(contentlet.getHost());
+        String userPromptTemplate = getAIUserTranslationPrompt(contentlet.getHost());
+
+        Context systemContext = VelocityUtil.getBasicContext();
+        systemContext.put("sourceLanguage", sourceLang.getLanguage() + "(" + sourceLang.getCountry() + ")");
+        systemContext.put("targetLanguage", targetLanguage.getLanguage() + "(" + targetLanguage.getCountry() + ")");
+
+        if (!translationKeysJSON.isEmpty()) {
+            systemContext.put("translationKeys", translationKeysJSON.toString());
+        }
+        String systemPrompt = Try.of(() -> VelocityUtil.eval(systemPromptTemplate, systemContext))
```

Comment:
Prompting will be more common in the future, a Service to encapsulate their generation wouldnt be bad